app/models/user.rb

```diff
@@ -1,3 +1,19 @@
 class User < ApplicationRecord
-  enum role: [:user, :admin]
+  before_save :email_downcase
+  validates :name, presence: true, length: { maximum: Settings.val_name_max }
+  validates :email, presence: true, length: { maximum: Settings.val_email_max },
+                    format: { with: Settings.email_regex },
+                    uniqueness: { case_sensitive: false }
+  validates :password, presence: true,
+                       length: { minimum: Settings.val_pass_min }
+  validates :phone, presence: true, length: { maximum: Settings.val_phone_max },
+                    format: { with: Settings.phone_regex }
+  has_secure_password
+  enum role: %i[user admin]
```

Comment:
nên dùng dạng hash để ko bị phụ thuộc vào vị trí trong mảng
```suggestion
  enum role: %i{user: 0, admin: 1}
```